apps/src/labs/labRedux.ts

```diff
@@ -3,18 +3,103 @@
 // other existing redux store.  It also holds useful state for newer labs
 // that use LabContainer.
 
-import {createSlice, PayloadAction} from '@reduxjs/toolkit';
+import {
+  AnyAction,
+  createAsyncThunk,
+  createSlice,
+  PayloadAction,
+  ThunkDispatch,
+} from '@reduxjs/toolkit';
+import {Channel, ProjectManagerStorageType, Source} from './types';
+import LabRegistry from './LabRegistry';
+import ProjectManagerFactory from './projects/ProjectManagerFactory';
+import {
+  setProjectUpdatedAt,
+  setProjectUpdatedError,
+  setProjectUpdatedSaving,
+  setProjectUpdatedSaved,
+} from '../code-studio/projectRedux';
+import ProjectManager from './projects/ProjectManager';
 
-interface LabState {
+export interface LabState {
```

Comment:
mind adding docs to these fields?

Reply:
I added comments to the fields that could be confusing, lmk if you were thinking of any other documentation that was needed.